stylesheets/_component.rules.scss

```diff
@@ -298,3 +298,306 @@ td.table__form .rules + .form__group:last-of-type {
         width: auto;
     }
 }
+
+/**
+ * The following version 2 was developed to meet the needs of XFP conditional logic rules
+ */
+.rules-container--v2 {
+    border-left: 4px solid color(gray, lightest);
+    padding-left: 20px;
+    position: relative;
+
+    &::before,
+    &::after {
+        background-color: color(white);
+        content: '';
+        display: block;
+        height: 35px;
+        left: -4px;
+        position: absolute;
+        width: 24px;
+    }
+
+    &::before {
+        border-bottom: 4px solid color(gray, lightest);
+        top: 0;
+    }
+
+    &::after {
+        border-top: 4px solid color(gray, lightest);
+        bottom: 0;
+    }
+
+    .rule-links button {
+        border: 1px solid color(border);
+        border-radius: 5px;
+        display: inline-block;
+        margin-left: 25px;
+        min-width: 50px;
+        padding: 8px 0;
+        text-align: center;
+
+        &:hover {
+            background-color: color(background, light);
+        }
+    }
+
+    .rule-link {
+        position: relative;
+        padding-top: 10px;
+        padding-bottom: 10px;
+
+        &::after {
+            color: #dedede;
+            content: '\f061';
+            font-family: 'FontAwesome';
+            font-size: 19px;
+            left: 0;
+            position: absolute;
+            top: 17px;
+        }
+    }
+
+    .rule-link--or {
+        &::before {
+            content: '';
+            display: inline-block;
+            border-left: 4px solid color(gray, lightest);
+            margin-right: -5px;
+            height: 45px;
+            position: absolute;
+            top: -14px;
+            width: 0;
+        }
+
+        &::after {
+            left: 3px;
+        }
+
+        button {
+            background-color: lighten(#9d7ee3, 25);
+        }
+    }
+
+    .rule-block--else-if + .rule-links .rule-link--or,
+    .rule-block--and + .rule-links .rule-link--or {
+        border-left: 4px solid color(gray, lightest);
+        margin-left: 20px;
+        padding-left: 40px;
+
+        &::after {
+            left: 44px;
+        }
+    }
+
+    .rule-link--or + .rule-link--and {
+        padding-top: 0;
+
+        &::after {
+            top: 7px;
+        }
+    }
+
+    .rule-link--and {
+        border-left: 4px solid color(gray, lightest);
+        margin-left: 20px;
+
+        button {
```

Comment:
I know we aren't fully BEM compliant at times, but it does feel odd to see a mix of BEM `.thing--modifier` mixed with inheritance and element selectors in the same block.